src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.company.profile.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.company.profile.service.CompanyProfileService;
+
+@RestController
+public class CompanyProfileController {
+
+    private final CompanyProfileService companyProfileService;
+
+    public CompanyProfileController(CompanyProfileService companyProfileService) {
+        this.companyProfileService = companyProfileService;
+    }
+
+    // TODO Update with newly generated CompanyProfile object once DSND-524 is completed
+
+    /**
+     * Retrieve a company profile using a company number.
+     *
+     * @param companyNumber the company number of the company
+     * @return company profile api
+     */
+    @GetMapping("/company/{company_number}")
+    public ResponseEntity<CompanyProfileApi> getCompanyProfile(
+            @PathVariable("company_number") String companyNumber) {
+        try {
+            return companyProfileService.get(companyNumber)
+                    .map(companyProfile -> new ResponseEntity<>(companyProfile, HttpStatus.OK))
+                    .orElse(ResponseEntity.notFound().build());
+        } catch (Exception exception) {
```

Comment:
Can we please have a centra response error handler which catches exceptions centrally instead of each controller having a catch all statement to return 500?

Reply:
Added a global exception handler through `@ControllerAdvice` and removed try/catch.
At this point since nothing else is specified in this ticket it will catch any exception and return a status 500, can be customised down the line on an exception-class basis.